x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/type/SqlDataTypeConverter.java

```diff
@@ -80,31 +81,25 @@ public static DataType commonType(DataType left, DataType right) {
             return DATETIME;
         }
         if (left == TIME) {
-            if (right == DATE) {
```

Comment:
Why these removals?

Reply:
They're not reachable: https://github.com/elastic/elasticsearch/pull/65145#discussion_r528549790